README.md

```
@@ -111,6 +111,79 @@ cargo run --bin=benchmark --release -- $TEST_CORPUS_PATH/Public-Tests/B01_synthe
+```
+
+#### How it works
+
+1. **Discover** — walks the results directory, finds cases with `Cargo.toml`, matches them with corpus test vectors
+2. **Build** — runs `cargo build --release` for each case in parallel
+3. **Prepare libs** — for library cases: copies runners from the corpus, sets up a shared cando2 dependency, ensures `crate-type = ["cdylib"]`, and builds the runner workspace
+4. **Test** — executes tests in parallel and writes `results.csv`
+
+#### Expected results directory layout
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)